### PR TITLE
Feature/mentees crud

### PR DESCRIPTION
**Pull request by ferror18**
This PR is to add search functionality to the mentees list inside the headmaster board. Aside from that it doesn't address anything else.

### CODE REVIEW

File: src/components/pages/Headmaster/HeadmasterDashboard.css [hunk 1]
```diff
@@ -59,6 +59,17 @@
 	height: 3rem;
 }
 
+#menteeTittle {
```
Comment (Jahteo) [inline src/components/pages/Headmaster/HeadmasterDashboard.css:62]: Title vs Tittle

File: src/components/pages/Login/Login.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { connect } from 'react-redux';
 
 import { Form, Input } from 'antd';
 
-import { login } from '../../../state/actions';
+import { login, checkToken } from '../../../state/actions';
```
Comment (Jahteo) [inline src/components/pages/Login/Login.js:6]: just as heads up, my next pr rearranges checkToken & how this login works. Loved how you got a solid working solution in the meantime though.

**Review by Jahteo — APPROVED** [review]
Liked it.

File: package.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
     "@okta/okta-react": "^3.0.2",
     "@okta/okta-signin-widget": "^4.1.2",
     "antd": "^4.4.3",
-    "axios": "^0.19.2",
+    "axios": "0.21.1",
```
Comment (jidelson) [inline package.json:12]: Looks like you got an updated version of axios in there, that is good.

File: src/components/pages/Headmaster/Mentees/Mentees.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from 'react';
```
Comment (jidelson) [inline src/components/pages/Headmaster/Mentees/Mentees.js:1]: This file looks great. Good job!

File: src/state/actions/index.js [hunk 5]
```diff
@@ -24,23 +30,34 @@ export const login = data => dispatch => {
...
     .then(res => {
       window.location.replace('/profile/');
     })
     .catch(err => console.dir(err));
 };
-
 export const fetchHeadmasterProfile = id => dispatch => {
-  axios // ! This should later become available through axiosWithAuth() only once we figure out the Auth with Stakeholder's backend
-    .get(`${baseURL}/headmaster/${id}`) // change this later
+  axiosWithAuth() // ! This should later become available through axiosWithAuth() only once we figure out the Auth with Stakeholder's backend
```
Comment (jidelson) [inline src/state/actions/index.js:59]: Good clarifying comment regarding where this will go later

File: src/state/reducers/headmasterReducer.js [hunk 6]
```diff
@@ -4,26 +4,45 @@ import {
...
+  FETCH_MENTEE_SUCCESS,
+  FETCH_MENTEE_FAILURE,
+  FETCH_MENTEE_START,
 } from '../actions/actionTypes';
 
 const initialState = {
   villageData: {},
   schoolData: [],
   headmasterProfile: '',
+  mentees: [],
```
Comment (jidelson) [inline src/state/reducers/headmasterReducer.js:16]: Good job remembering to update the state

**Review by jidelson — APPROVED** [review]
Looks awesome. Good job!!

**Pull request merged**